### PR TITLE
refactor(eslint-plugin): [sort-ngmodule-metadata-arrays]: properly handle multiple fixes at once

### PR DESCRIPTION
**Pull request by rafaelss95**
Besides addressing the issue #675, the report node is more accurate now and reports nested arrays.

PS: TBH, I'm not sure which scope should be this PR, so I used `refactor` 🤷‍♂️

Closes #677.

### CODE REVIEW

File: packages/eslint-plugin/src/rules/sort-ngmodule-metadata-arrays.ts [hunk 1]
```diff
@@ -16,42 +16,124 @@ export default createESLintRule<Options, MessageIds>({
...
       category: 'Best Practices',
       recommended: false,
+      suggestion: true,
     },
     fixable: 'code',
     schema: [],
     messages: {
       sortNgmoduleMetadataArrays:
         '`NgModule` metadata arrays should be sorted in ASC alphabetical order',
+      suggestFix: 'Sort members (WARNING! comments will probably be messed up)',
```
Comment (rafaelss95) [inline packages/eslint-plugin/src/rules/sort-ngmodule-metadata-arrays.ts:26]: Please, suggest a better message if that's the case.

File: packages/eslint-plugin/src/rules/sort-ngmodule-metadata-arrays.ts [hunk 2]
```diff
@@ -16,42 +16,124 @@ export default createESLintRule<Options, MessageIds>({
...
-          node: unorderedNode,
+          node: firstUnorderedNode,
           messageId: 'sortNgmoduleMetadataArrays',
-          fix: (fixer) => [
-            fixer.replaceText(unorderedNode, nextNode.name),
-            fixer.replaceText(nextNode, unorderedNode.name),
-          ],
+          ...(hasCommentsWithinElements(sourceCode, elements)
+            ? { suggest: [{ messageId: 'suggestFix', fix }] }
+            : { fix }),
```
Comment (rafaelss95) [inline packages/eslint-plugin/src/rules/sort-ngmodule-metadata-arrays.ts:56]: I even tried to use `fix`, even if there were comments, but for god's sake, besides being very costly to maintain such an implementation, it's also very prone to failure.

File: packages/utils/src/utils.ts [hunk 3]
```diff
@@ -65,3 +65,32 @@ export function capitalize<T extends string>(text: T): Capitalize<T> {
 export function withoutBracketsAndWhitespaces(text: string): string {
   return text.replace(/[[\]\s]/g, '');
 }
+
+/**
+ * Split the `array` into two `arrays`, where the first array contains all
```
Comment (rafaelss95) [inline packages/utils/src/utils.ts:70]:
```suggestion
 * Splits the `array` into two `arrays`, where the first array contains all
```

**Comment by JamesHenry** [comment]
Hi @rafaelss95 thanks so much for this, please see my latest email for how I'd love to see this broken down, please let me know on there if you have any questions at all!

**Pull request closed without merging**